automation/generate_ami.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+
+set -e
+
+IMAGE="$IMAGE"
```

Comment:
@kaisoz could you please make sure you pass a linter like shellcheck? There are a few cases were you need double quotes in variables to prevent word splitting and globbing.

Reply:
Yes, I did but it doesn't support the case where you return a variable as a reference. In my opinion, shellcheck sometimes is too strict because it also flags variables whose value, by definition cannot suffer such consequences (word splitting and globbing). An example is the BALENACLI_TOKEN variable. By definition, a token with a space is invalid, so if the token is split, is going to be invalid anyway.  

Adding double-quotes worsen readability in a language that is already difficult to read... and although this is not a great example, I think is simple enough to explain my point :smile:

Reply:
Does `shellcheck` give you a ref to the violation? If so, look this up and there is normally documented in the shellcheck docs. You can even ignore specific violations with a meta comment if all else fails.

Reply:
Having a clean shellcheck pass goes a long way to improve the maintainability of the script, so I'd agree with Anton that using the meta-comments to ignore specific instances is the best approach.